<launch>

  <!-- USE: roslaunch my_mira_description urdf_visualize.launch model:='$(find myrobot_package)/urdf/myrobot.urdf' -->
  <arg name="model" default="$(find underwater_sim)/urdf/underwater.urdf"/>


  <param name="robot_description" command="cat $(arg model)" />

  <!-- send fake joint values -->
  <node name="joint_state_publisher" pkg="joint_state_publisher" type="joint_state_publisher">
    <param name="use_gui" value="false"/>
  </node>

  
  <!-- Combine joint values -->
  <node name="robot_state_publisher" pkg="robot_state_publisher" type="state_publisher"/>
   
  <!-- connect to arduino -->
  <node name="arduino" pkg="rosserial_python" type="serial_node.py">
    <param name="port" value="/dev/ttyACM0"/>
  </node>
   
  <!--broadcast parameters-->
  <node name="imu_broadcast" pkg="underwater_sim" type="broadcast.py"/>
  <!-- Show in Rviz   -->
  <!--<node name="rviz" pkg="rviz" type="rviz" args="-d $(find my_mira_pkg)/rviz_config/urdf.rviz"/>-->
  <node name="rviz" pkg="rviz" type="rviz" args="-d $(find underwater_sim)/rviz/underwater.rviz"/>

</launch>
